spec/requests/v0/health_care_applications_spec.rb

```diff
@@ -453,46 +453,58 @@ def self.expect_async_submit
+            allow_any_instance_of(HCA::Service).to receive(:submit_form) do
+              raise error
+            end
           end
-        end
 
-        context 'with a validation error' do
-          let(:error) { HCA::SOAPParser::ValidationError.new }
+          context 'with a validation error' do
+            let(:error) { HCA::SOAPParser::ValidationError.new }
 
-          it 'renders error message' do
-            subject
+            it 'renders error message' do
+              subject
 
-            expect(response).to have_http_status(:unprocessable_entity)
-            expect(JSON.parse(response.body)).to eq(
-              'errors' => [
-                { 'title' => 'Operation failed', 'detail' => 'Validation error', 'code' => 'HCA422', 'status' => '422' }
-              ]
-            )
+              expect(response).to have_http_status(:unprocessable_entity)
+              expect(JSON.parse(response.body)).to eq(
+                'errors' => [
+                  { 'title' => 'Operation failed', 'detail' => 'Validation error', 'code' => 'HCA422', 'status' => '422' }
+                ]
+              )
+            end
           end
-        end
 
-        context 'with a SOAP error' do
-          let(:error) { Common::Client::Errors::HTTPError.new('error message') }
+          context 'with a SOAP error' do
+            let(:error) { Common::Client::Errors::HTTPError.new('error message') }
 
-          before do
-            allow(Settings.sentry).to receive(:dsn).and_return('asdf')
-          end
+            before do
+              allow(Settings.sentry).to receive(:dsn).and_return('asdf')
+            end
 
-          it 'renders error message' do
-            expect(Sentry).to receive(:capture_exception).with(error, level: 'error').once
+            it 'renders error message' do
+              expect(Sentry).to receive(:capture_exception).with(error, level: 'error').once
 
-            subject
+              subject
 
-            expect(response).to have_http_status(:bad_request)
-            expect(JSON.parse(response.body)).to eq(
-              'errors' => [
-                { 'title' => 'Operation failed', 'detail' => 'error message', 'code' => 'VA900', 'status' => '400' }
-              ]
-            )
+              expect(response).to have_http_status(:bad_request)
+              expect(JSON.parse(response.body)).to eq(
+                'errors' => [
+                  { 'title' => 'Operation failed', 'detail' => 'error message', 'code' => 'VA900', 'status' => '400' }
+                ]
+              )
+            end
+          end
+        end
+
+        context "when the 'va1010_forms_enrollment_system_service_enabled' flipper is disabled" do
+          before do
+            Flipper.disable(:va1010_forms_enrollment_system_service_enabled)
+            test_veteran.delete('email')
+            allow_any_instance_of(HCA::Service).to receive(:post) do
+              raise error
+            end
```

Comment:
This allow block was causing tests to fail. Can someone explain why we were using post as the receive method for the HCA:Service instead of simply using submit_form? I'm genuinely curious. In this instance, is there a benefit to using an inherited method as opposed to a method native to the class?

Reply:
What is this block actually testing? I see we delete the email form the `test_veteran`, but do we do anything else here? Maybe I'm just misunderstanding.